### PR TITLE
Updated javascript/html/view/form for adminactionform

### PR DESCRIPTION
**Pull request by pault2k14**

### CODE REVIEW

File: its/items/tests.py [hunk 1]
```diff
@@ -829,11 +829,33 @@ def test_clean_with_errors(self):
 
         with patch('its.items.forms.AdminActionForm.clean', return_value=data):
                 with patch("its.items.forms.AdminActionForm.add_error") as add_error:
-                    form = AdminActionForm(data, user=user)
+                    form = AdminActionForm(data, current_user=user)
                     form.clean()
 
                     add_error.assert_any_call_with("note", "Note required when choosing action of type Other.")
 
+        # Test 3 - Check for errors when action_choice not in dictionary.
```
Comment (mdj2) [inline its/items/tests.py]: I plugged this test into your old code (I had to change `current_user` to `user`) and it passed. This new test doesn't detect the bug.

File: its/items/forms.py [hunk 2]
```diff
@@ -59,18 +59,20 @@ class AdminActionForm(forms.Form):
...
-    def __init__(self, *args, user, **kwargs):
+        
+    def __init__(self, *args, current_user, **kwargs):
 
         """
         Allow only the return option to be selected by lab attendants.
         """
 
-        self.user = user
+        self.user = current_user
```
Comment (mdj2) [inline its/items/forms.py]: Not sure what this change was about...

File: its/items/forms.py [hunk 3]
```diff
@@ -101,12 +103,12 @@ def clean(self):
...
 
         cleaned_data = super().clean()
-        action_choice = cleaned_data.get("action_choice")
+        action_choice = self.cleaned_data.get("action_choice", Action.objects.get(machine_name=Action.RETURNED))
         note = cleaned_data.get("note")
         first_name = cleaned_data.get("first_name")
         last_name = cleaned_data.get("last_name")
         email = cleaned_data.get("email")
-
+
```
Comment (mdj2) [inline its/items/forms.py]: flake8 violation.

File: its/items/forms.py [hunk 4]
```diff
@@ -101,12 +103,12 @@ def clean(self):
         """
 
         cleaned_data = super().clean()
-        action_choice = cleaned_data.get("action_choice")
+        action_choice = self.cleaned_data.get("action_choice", Action.objects.get(machine_name=Action.RETURNED))
```
Comment (mdj2) [inline its/items/forms.py]: You duplicate this line in your `save` method. You can get around that by doing `cleaned_data['action_choice'] = cleaned_data.get("action_choice", Action.objects.get(machine_name=Action.RETURNED))`. Then in your save method, you can just access `cleaned_data['action_choice']`

File: its/items/forms.py [hunk 5]
```diff
@@ -131,12 +133,12 @@ def save(self, *args, item_pk, current_user, **kwargs):
...
         """
 
-        action_choice = self.cleaned_data["action_choice"]
         item = Item.objects.get(pk=item_pk)
+        action_choice = self.cleaned_data.get("action_choice", Action.objects.get(machine_name=Action.RETURNED))
         first_name = self.cleaned_data.get("first_name")
         last_name = self.cleaned_data.get("last_name")
         email = self.cleaned_data.get("email")
-        new_action = Action.objects.get(name=self.cleaned_data['action_choice'])
+        new_action = Action.objects.get(name=action_choice)
```
Comment (mdj2) [inline its/items/forms.py]: This line is unnecessary and could bug out if you ever change `Action.__str__`. At this point in the program, `self.cleaned_data['action_choice']` is already an `Action` model, so you don't need to fetch it from the DB again.

File: its/items/tests.py (hunk 1 above)
Comment (pault2k14) [inline its/items/tests.py]: That's why the comment says "Still working on this test."

**Pull request merged**